pkg/sender/in_cluster.go

```diff
@@ -44,9 +51,11 @@ func (i *inClusterSender) Send(ctx context.Context, ce cloudevents.Event) error
 			},
 		},
 		Spec: batchv1.JobSpec{
+			BackoffLimit: retries(),
 			Template: corev1.PodTemplateSpec{
 				Spec: corev1.PodSpec{
-					RestartPolicy: corev1.RestartPolicyNever,
+					RestartPolicy:         corev1.RestartPolicyNever,
+					ActiveDeadlineSeconds: deadline(),
```

Comment:
Using this deadline should be safe (the image download time isn't counting into this deadline), as checked here: https://drive.google.com/file/d/19js79hNTqPJG7CogHH7O6-CRSijDXRNE/view